### PR TITLE
(maint) Update repo definitions to use non-deprecated methods

### PR DESCRIPTION
**Pull request by melissa**

### CODE REVIEW

File: configs/platforms/el-7-x86_64.rb [hunk 1]
```diff
@@ -3,7 +3,8 @@
   plat.defaultdir "/etc/sysconfig"
   plat.servicetype "systemd"
 
-  plat.provision_with "yum install --assumeyes autoconf automake createrepo rsync gcc make rpmdevtools rpm-libs yum-utils rpm-sign;echo '[pl-build-tools]\nname=pl-build-tools\ngpgcheck=0\nbaseurl=http://pl-build-tools.delivery.puppetlabs.net/yum/el/6/$basearch' > /etc/yum.repos.d/pl-build-tools.repo"
```
Comment (melissa) [inline configs/platforms/el-7-x86_64.rb]: Notice this was pointing to an el 6 repo. That was my fault.

**Comment by melissa** [comment]
Testing at http://jenkins-release.delivery.puppetlabs.net/job/vanagon_generic_job/262/ (ignore the el-7-s390 failure)

**Comment by melissa** [comment]
Everything but el-7-s390 passed http://jenkins-release.delivery.puppetlabs.net/job/vanagon_generic_job/262/

**Comment by underscorgan** [comment]
:+1:

**Comment by melissa** [comment]
Waiting for tags/release stuff to finish

cc/ @McdonaldSeanp

File: configs/platforms/el-4-i386.rb [hunk 2]
```diff
@@ -4,7 +4,9 @@
   plat.servicetype "sysv"
   plat.tar "/opt/pl-build-tools/bin/tar"
 
-  plat.provision_with "echo '[build-tools]\nname=build-tools\ngpgcheck=0\nbaseurl=http://enterprise.delivery.puppetlabs.net/build-tools/el/4/$basearch' > /etc/yum.repos.d/build-tools.repo;echo '[pl-build-tools]\nname=pl-build-tools\ngpgcheck=0\nbaseurl=http://pl-build-tools.delivery.puppetlabs.net/yum/el/4/$basearch' > /etc/yum.repos.d/pl-build-tools.repo;yum install -y autoconf automake createrepo rsync gcc make rpm-build rpm-libs yum-utils pl-tar; yum update -y pkgconfig"
+  plat.add_build_repository "http://pl-build-tools.delivery.puppetlabs.net/yum/pl-build-tools-release-el-4.noarch.rpm"
+  plat.provision_with "echo '[build-tools]\nname=build-tools\ngpgcheck=0\nbaseurl=http://enterprise.delivery.puppetlabs.net/build-tools/el/4/$basearch' > /etc/yum.repos.d/build-tools.repo;yum install -y autoconf automake createrepo rsync gcc make rpm-build rpm-libs yum-utils pl-tar; yum update -y pkgconfig"
```
Comment (stahnma) [inline configs/platforms/el-4-i386.rb]: Is this pointing to build-tools and pl-build-tools?

File: configs/platforms/el-5-i386.rb [hunk 3]
```diff
@@ -3,7 +3,8 @@
   plat.defaultdir "/etc/sysconfig"
   plat.servicetype "sysv"
 
-  plat.provision_with "yum install -y --nogpgcheck autoconf automake createrepo rsync gcc make rpmdevtools rpm-libs yum-utils rpm-sign rpm-build; echo '[build-tools]\nname=build-tools\nbaseurl=http://enterprise.delivery.puppetlabs.net/build-tools/el/5/$basearch' > /etc/yum.repos.d/build-tools.repo;echo '[pl-build-tools]\nname=pl-build-tools\ngpgcheck=0\nbaseurl=http://pl-build-tools.delivery.puppetlabs.net/yum/el/5/$basearch' > /etc/yum.repos.d/pl-build-tools.repo"
+  plat.add_build_repository "http://pl-build-tools.delivery.puppetlabs.net/yum/pl-build-tools-release-el-5.noarch.rpm"
+  plat.provision_with "yum install -y --nogpgcheck autoconf automake createrepo rsync gcc make rpmdevtools rpm-libs yum-utils rpm-sign rpm-build; echo '[build-tools]\nname=build-tools\nbaseurl=http://enterprise.delivery.puppetlabs.net/build-tools/el/5/$basearch' > /etc/yum.repos.d/build-tools.repo"
```
Comment (stahnma) [inline configs/platforms/el-5-i386.rb]: Same.

File: configs/platforms/el-7-x86_64.rb [hunk 4]
```diff
@@ -3,7 +3,8 @@
...
+  plat.add_build_repository "http://pl-build-tools.delivery.puppetlabs.net/yum/pl-build-tools-release-el-7.noarch.rpm"
```
Comment (stahnma) [inline configs/platforms/el-7-x86_64.rb]: We could change it to #{plat.get_os_version} for portability.

File: configs/platforms/el-4-i386.rb (hunk 2 above)
Comment (melissa) [inline configs/platforms/el-4-i386.rb]: @stahnma we're enabling the repo at `http://enterprise.delivery.puppetlabs.net/build-tools/el/4/$basearch`. There aren't any .repo or release rpms for these repos.

File: configs/platforms/el-7-x86_64.rb (hunk 4 above)
Comment (melissa) [inline configs/platforms/el-7-x86_64.rb]: did it!

**Comment by melissa** [comment]
Testing again at http://jenkins-release.delivery.puppetlabs.net/job/vanagon_generic_job/265/

**Comment by pyther** [comment]
+1 Was going to submit a patch like this, you beat me too it!

**Comment by shermdog** [comment]
I :heart:  this!

**Comment by pyther** [comment]
Is there any plan to merge this into master?

**Comment by melissa** [comment]
@pyther there are regular merge ups from stable to master. It should get into master within the next week or two. I can't guarentee when exactly the merge up will happen though

**Comment by pyther** [comment]
@melissa Thanks for the clarification. In my mind, I thought the workflow was such that patches, such as this one, get merged into master and then master get merged into stable.

**Comment by melissa** [comment]
Sure thing!

**Pull request merged**